src/pal/src/exception/seh.cpp

```diff
@@ -262,6 +282,9 @@ SEHProcessException(PAL_SEHException* exception)
 
         if (CatchHardwareExceptionHolder::IsEnabled())
         {
+            if(exceptionRecord->ExceptionFlags | EXCEPTION_ON_STACK)
```

Comment:
To simplify the callers, it would be nice to move the check into the actual function that updates the exception. For exception that already has records on heap, the function would become a no-op.